java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -55,14 +58,12 @@ public static ListVector empty(String name, BufferAllocator allocator) {
     return new ListVector(name, allocator, FieldType.nullable(ArrowType.List.INSTANCE), null);
   }
 
-  final UInt4Vector offsets;
-  final BitVector bits;
-  private final List<BufferBacked> innerVectors;
-  private Mutator mutator = new Mutator();
-  private Accessor accessor = new Accessor();
+  private ArrowBuf validityBuffer;
   private UnionListReader reader;
   private CallBack callBack;
   private final FieldType fieldType;
+  private int validityAllocationSizeInBytes;
+  private int lastSet;
```

Comment:
Similarly, does it make sense to move this to base class?

Reply:
https://github.com/apache/arrow/pull/1203#discussion_r146081974

Reply:
Ok this is fine.